core/commands/swarm.go

```diff
@@ -476,6 +507,21 @@ func peersWithAddresses(addrs []string) (pis []pstore.PeerInfo, err error) {
 	return pis, nil
 }
 
+func getNode(req cmds.Request, res cmds.Response) *core.IpfsNode {
```

Comment:
this should probably be `getOnlineNode` or something, and lets go ahead and return the errors and handle them in the caller.